File: js/05_smallest_multiple.js
// 2520 is the smallest number that can be divided by each of
// the numbers from 1 to 10 without any remainder.
// What is the smallest positive number that is evenly divisible
// by all of the numbers from 1 to 20?
var timer = function(name) {
	var start = new Date();
	return {
		stop: function() {
			var end  = new Date();
			var time = end.getTime() - start.getTime();
			console.log('Timer:', name, 'finished in', time, 'ms');
		}
	}
};


function smallest(n) {
	var t = timer(n)
	let start = n;
	let num = start;
	while (num > 0) {
		if (n % num == 0) {
			num -= 1;
		} else {
			num = start;
		}
		if (num == 0) {
			break;
		} else {
			n += 1;
		}

	}
	console.log(n + 1);
	t.stop();
}

smallest(10);
smallest(20);
smallest(22);

File: C/05_smallest_multiple.c
#include <stdio.h>
#include <time.h>

int main(void)
{
	clock_t start, end;
	double cpu_time_used;

	start = clock();

	long long n;
	printf("Please input an integer: ");
	scanf("%lld", &n);
	int begin = n;
	int num = begin;
	while (num > 0) {
		if (n % num == 0) {
			num -= 1;
		} else {
			num = begin;
		}
		if (num == 0) {
			break;
		} else {
			n += 1;
		}
	}
	printf("> %lld\n", n + 1);
	end = clock();
	cpu_time_used = ((double) (end-start)) / CLOCKS_PER_SEC;
	printf("Timer: %i finished in %f s\n", begin, cpu_time_used);
}

	

